torch_sim/models/metatensor.py

```diff
@@ -0,0 +1,328 @@
+"""Wrapper for metatensor-based models in TorchSim.
+
+This module provides a TorchSim wrapper of metatensor models for computing
+energies, forces, and stresses for atomistic systems, including batched computations
+for multiple systems simultaneously.
+
+The MetatensorModel class adapts metatensor models to the ModelInterface protocol,
+allowing them to be used within the broader torch_sim simulation framework.
+
+Notes:
+    This module depends on the metatensor-torch package.
+"""
+
+from collections.abc import Callable
+from pathlib import Path
+
+import torch
+
+from torch_sim.models.interface import ModelInterface
+from torch_sim.neighbors import vesin_nl_ts
+from torch_sim.state import SimState, StateDict
+
+
+try:
+    from metatensor.torch import Labels, TensorBlock
+    from metatensor.torch.atomistic import (
+        ModelEvaluationOptions,
+        ModelOutput,
+        System,
+        load_atomistic_model,
+    )
+    from metatrain.utils.io import load_model
+
+except ImportError:
+
+    class MetatensorModel(torch.nn.Module, ModelInterface):
+        """Metatensor model wrapper for torch_sim.
+
+        This class is a placeholder for the MetatensorModel class.
+        It raises an ImportError if metatensor is not installed.
+        """
```

Comment:
To work with the documentation build this should define a dummy __init__ method too, something like: 

```suggestion
    class MetatensorModel(torch.nn.Module, ModelInterface):
        """Metatensor model wrapper for torch_sim.

        This class is a placeholder for the MetatensorModel class.
        It raises an ImportError if metatensor is not installed.
        """
        def __init__():
            raise ImportError("metatensor must be installed to use this model.")
```

Reply:
Done, thanks!